### ISSUE TITLE
Process fails if I throw error in the hook or step

### ISSUE THREAD

**Issue by klokovas** [mucsi96/nightwatch-cucumber#132]
**I'm submitting a ...**  (check one with "x")

```
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead use https://groups.google.com/d/forum/nightwatch-cucumber
```

**Current behavior**
I have hooks.js. I want use assertion after each step. I check browser logs.

```
module.exports = function () {
  this.AfterStep(function (features,  callback) {
    throw 'err';
    callback();
  });
}
```

OR

```
module.exports = function () {
  this.AfterStep(function (features) {
    this.assert.someNWassertion();
  });
}
```

When I run the tests I get an error:

```
Unhandled Rejection at: Promise Promise {
  _c: [],
  _a: [],
  _s: 2,
  _d: true,
  _v: 'node_modules/cucumber/lib/cucumber/support_code/library.js:17 node_modules/cucumber/lib/cucumber/support_code/library.js:17 err',
  _h: 0,
  _n: false } reason: node_modules/cucumber/lib/cucumber/support_code/library.js:17 node_modules/cucumber/lib/cucumber/support_code/library.js:17 err
```

**Please tell us about your environment:**
- **nightwatch-cucumber version:**
  `5.5.2`
- **nightwatch version:**
  `0.9.8`
- **cucumber version:**
  `1.3.1`
- **Node.js version:**
  `6.3.1`

**Comment by klokovas** [comment]
If I have promise in the step and it throws error then I have `Unhandled Rejection` too.

```
this.Given(/^any step$/, function() {
    return anyPromise().then(() => {...});
})
```

output:
```
  ✔ Given any step
Unhandled Rejection at: Promise Promise {
  _bitField: 18087936,
  _fulfillmentHandler0: 'some error',
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined } reason: some error
```

**Comment by mucsi96** [comment]
Hi @klokovas! Thanks for you feedback! I should be fixed in new release `5.6.0`. Can you update and provide a feedback again?

**Comment by klokovas** [comment]
Hi @mucsi96 !

```
[X] It works for step definitions.
[ ] But this doesn't work for hooks.
[ ] And now you have an error in the documentation: hooks always provide Nightwatch API.
```

**Comment by mucsi96** [comment]
Hi @klokovas! Thanks for the report! I will take a look at it

**Comment by mucsi96** [comment]
Hi @klokovas! I checked the issue. There is an issue in cucumber.js. The problem is here: https://github.com/cucumber/cucumber-js/blob/master/lib/cucumber/runtime/event_broadcaster.js#L30

Throwing an error in an asynchronous function cause unhandled errors and unhandled Rejections. I tried to fix that locally by replacing the `throw error;` with `return callback(error)`. But in that case Cucumber just ignores the event handler error and does not log any error related information to the console. :(
I you wish rise a ticket there. I hope that helps. I am closing this issue.
